posydon/utils/common_functions.py

```diff
@@ -2620,18 +2620,18 @@ def calculate_Mejected_for_integrated_binding_energy(profile, Ebind_threshold,
     i = 0
     Ebind_so_far = 0.0 # the integration from surface going inwards of the binding energy (negative in principle)
 
-    while abs(Ebind_so_far) < Ebind_threshold:
+    while (abs(Ebind_so_far) < Ebind_threshold) and (i<len(donor_mass)):
         Grav_energy_of_cell = (-const.standard_cgrav * donor_mass[i]
                                * const.Msun * donor_dm[i]*const.Msun
                                / (donor_radius[i]*const.Rsun))
         # integral of gravitational energy as we go deeper into the star
         Grav_energy = Grav_energy + Grav_energy_of_cell
         U_energy = U_energy + specific_internal_energy[i]*donor_dm[i]*const.Msun
         Ebind_so_far = Grav_energy + factor_internal_energy * U_energy
-        ind_threshold = i
         i=i+1
+    ind_threshold = i
```

Comment:
Because you moved it now behind the increment of `i` it should read `ind_threshold = i-1`